back/app/services/json_forms_service.rb

```diff
@@ -41,75 +41,39 @@ def user_ui_and_json_multiloc_schemas(fields)
     }
   end
 
-  def input_ui_and_json_multiloc_schemas(fields, current_user, input_term)
+  def input_ui_and_json_multiloc_schemas(fields, current_user, participation_method, input_term)
     return if fields.empty?
 
-    participation_context = fields.first.resource.participation_context
-    supports_answer_visible_to = Factory.instance.participation_method_for(participation_context).supports_answer_visible_to?
-
-    visible_fields = fields.reject do |field|
-      !field.enabled? || field.hidden?
-    end
-    json_schema_multiloc = InputJsonSchemaGeneratorService.new.generate_for visible_fields
-    ui_schema_multiloc = InputUiSchemaGeneratorService.new(input_term, supports_answer_visible_to).generate_for visible_fields
+    fields.reject!(&:hidden?)
```

Comment:
I don't completely understand the changes here. Before, were we rejecting fields if they were not enable or hidden? And now we're only rejecting them if they're hidden? :thinking:

Reply:
So before, we filtered out these values and assigned the remaining fields to "visible_fields", but we don't need to do that anymore then?

Reply:
Yes indeed, actually I think this service should never do any filtering on the fields and should only care about generating the corresponding schemas for whichever fields are passed. Then it's the caller's responsibility to pass the right fields. In the end, I added back `fields.reject!(&:hidden?)` because this is more of an explicit requirement (for inputs, the hidden attribute is only used to hide their fields from the form/schema), but I hope one day we can remove this line.

An alternative could be to pass the custom form, and let the JsonFormsService determine the fields, but that would be less consistent with the registration (user) fields, and I hope that we can unify the code of the schema's for both one day.

Reply:
Ok gotchya, thank you! :)